src/OrchardCore.Modules/OrchardCore.ReCaptcha/Users/Handlers/LoginFormEventEventHandler.cs

```diff
@@ -1,17 +1,26 @@
+using Microsoft.AspNetCore.Identity;
 using OrchardCore.ReCaptcha.Services;
+using OrchardCore.Users;
 using OrchardCore.Users.Events;
 
 namespace OrchardCore.ReCaptcha.Users.Handlers;
 
 public sealed class LoginFormEventEventHandler : LoginFormEventBase
 {
     private readonly ReCaptchaService _reCaptchaService;
+    private readonly SignInManager<IUser> _signInManager;
 
-    public LoginFormEventEventHandler(ReCaptchaService reCaptchaService)
+    public LoginFormEventEventHandler(ReCaptchaService reCaptchaService, SignInManager<IUser> signInManager)
     {
         _reCaptchaService = reCaptchaService;
+        _signInManager = signInManager;
     }
 
-    public override Task LoggingInAsync(string userName, Action<string, string> reportError)
-        => _reCaptchaService.ValidateCaptchaAsync(reportError);
+    public override async Task LoggingInAsync(string userName, Action<string, string> reportError)
+    {
+        if (await _signInManager.GetExternalLoginInfoAsync() == null)
```

Comment:
Same as below, I would invert the condition and add a return. Also add a small comment.